sources/packages/api/src/services/student-account-applications/student-account-applications.service.ts

```diff
@@ -0,0 +1,197 @@
+    auditUserId: number,
+  ) {
+    const accountApplication = await this.repo.findOne({
+      select: { id: true, user: { id: true } },
+      relations: { user: true },
+      where: { id },
+    });
+
+    if (!accountApplication) {
+      throw new CustomNamedError(
+        "Student account application not found.",
+        STUDENT_ACCOUNT_APPLICATION_NOT_FOUND,
+      );
+    }
+
+    const userInfo = {
+      userId: accountApplication.user.id,
+      lastName: studentProfile.lastName,
+      givenNames: studentProfile.firstName,
+      email: studentProfile.email,
+      birthdate: getISODateOnlyString(studentProfile.dateOfBirth),
+      gender: studentProfile.gender,
+    } as UserInfo;
+
+    const studentInfo = {
+      phone: studentProfile.phone,
+      sinNumber: studentProfile.sinNumber,
+      addressLine1: studentProfile.addressLine1,
+      addressLine2: studentProfile.addressLine2,
+      provinceState: studentProfile.provinceState,
+      country: studentProfile.country,
+      city: studentProfile.city,
+      postalCode: studentProfile.postalCode,
+      selectedCountry: studentProfile.selectedCountry,
+    };
+
+    return this.dataSource.transaction((entityManager) => {
+      // Update the student account application with the approval.
+      const auditUser = { id: auditUserId } as User;
+      accountApplication.assessedBy = auditUser;
+      accountApplication.modifier = auditUser;
+      accountApplication.assessedDate = new Date();
+      entityManager
+        .getRepository(StudentAccountApplication)
+        .save(accountApplication);
+      // Create the new student and updates the user.
+      return this.studentService.createStudentFromAccountApplication(
+        userInfo,
+        studentInfo,
+        auditUserId,
+        entityManager,
+        accountApplication.id,
+      );
+    });
+  }
+
+  /**
+   * Declines the student account application.
+   * @param id student account application id.
+   */
+  async declineStudentAccountApplication(id: number): Promise<void> {
+    // TODO: To be changed to a soft delete.
+    const deleteResult = await this.repo
+      .createQueryBuilder()
+      .delete()
+      .from(StudentAccountApplication)
+      .where({ id })
+      .execute();
+    if (deleteResult.affected === 0) {
+      throw new CustomNamedError(
+        "Student account application not found.",
+        STUDENT_ACCOUNT_APPLICATION_NOT_FOUND,
+      );
+    }
+  }
+
+  /**
+   * Checks if the user has a pending student account applications.
+   * @param userId user to be verified.
+   * @returns true, if there is a pending student account application,
+   * otherwise, false.
+   */
+  async hasPendingStudentAccountApplication(userId: number): Promise<boolean> {
+    const accountApplication = await this.repo.findOne({
+      select: {
+        id: true,
+      },
+      where: {
+        assessedDate: IsNull(),
+        user: {
+          id: userId,
```

Comment:
in this case,` relations: { user: true },` not needed?

Reply:
While using it only in the `where` clause it was not needed. If selecting something from the user then it does not work without adding it.